### PR TITLE
bot: dependabot workflow automation for updating dependency

### LINKED ISSUE #11215: dependabot workflow automation for updating dependency

### PR DESCRIPTION
**Pull request by Rajpratik71**
**Description of your changes:**

Dependabot workflow automation for updating dependency

**Which issue is resolved by this Pull Request:**
Resolves #11215

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

Signed-off-by: Pratik Raj <rajpratik71@gmail.com>

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: 2
+updates:
+  # Dependencies listed in go.mod
+  - package-ecosystem: "gomod"
```
Comment (travisn) [inline .github/dependabot.yml:4]: I'm curious to see how well this automatic update works. Updating go modules sometimes involves other code changes that come with the new modules.

**Review by travisn — CHANGES_REQUESTED** [review]
For the commit message how about a prefix of `bot:`? The prefix `feat` is not recognized and is failing the commit linter.

File: .github/dependabot.yml (hunk 1 above)
Comment (Rajpratik71) [reply .github/dependabot.yml:4]: @travisn  Yes in the general development scenario developer will update dependencies but assuming in other scenarios there are updates in upstream dependencies, in those cases this automation will help.

**Comment by Rajpratik71** [comment]
> For the commit message how about a prefix of `bot:`? The prefix `feat` is not recognized and is failing the commit linter.

yes we can go for `bot:` or `automation:`.

Does this make sense?

**Comment by travisn** [comment]
> > For the commit message how about a prefix of `bot:`? The prefix `feat` is not recognized and is failing the commit linter.
> 
> yes we can go for `bot:` or `automation:`.
> 
> Does this make sense?

Looks good with `bot:`, the prefix just needs to be in [this list](https://github.com/rook/rook/blob/master/.commitlintrc.json).

File: .github/dependabot.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    directory: "/" # Location of package manifests
+    schedule:
+      interval: "weekly"
+
+  # Dependencies listed in .github/workflows/*.yml
+  - package-ecosystem: "github-actions"
```
Comment (subhamkrai) [inline .github/dependabot.yml:10]: with these we'll also get the update for github action, earlier we're only getting update for go.mod. right?

**Review by subhamkrai — APPROVED** [review]
LGTM!

**Review by travisn — APPROVED** [review]

File: .github/dependabot.yml (hunk 2 above)
Comment (travisn) [reply .github/dependabot.yml:10]: I don't remember seeing them before for either type

Comment (Rajpratik71) [reply .github/dependabot.yml:10]: yes this workflow will also update github-actions

**Pull request merged**